### PR TITLE
Private utility function documentation updates, round 5.

### PR DESCRIPTION
**Pull request by mmathesius**
A (hopefully!) final batch of documentation updates for private functions towards the resolution of #317.

### CODE REVIEW

File: modulemd/include/private/modulemd-yaml.h [hunk 1]
```diff
@@ -692,11 +948,45 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
...
+ * EMIT_SCALAR_FULL:
+ * @emitter: (inout): A libyaml emitter object positioned where a scalar
+ * belongs in the YAML document.
+ * @error: (out): A #GError that will return the reason for an output error.
+ * @value: (in): The scalar (string) to be written.
+ * @style: (in): The YAML scalar style for the output.
+ *
+ * Emits scalar @value using style @style.
+ *
+ * DIRECT USE OF THIS MACRO SHOULD BE AVOIDED. This is part of the internal
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: This one is actually okay to use directly. It's intended to allow code to specify something other than `YAML_PLAIN_SCALAR_STYLE` (such as the `YAML_FOLDED_SCALAR_STYLE` for the `data.description` section).

File: modulemd/include/private/modulemd-yaml.h [hunk 2]
```diff
@@ -402,12 +632,14 @@ mmd_emitter_scalar (yaml_emitter_t *emitter,
 
 /**
  * mmd_emitter_strv:
- * @emitter: (intout): A yaml emitter object positioned at the beginning of a value place to emit a sequence.
+ * @emitter: (inout): A libyaml emitter object positioned at the start of where
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: Good catch.

File: modulemd/include/private/modulemd-translation-entry-private.h [hunk 3]
```diff
@@ -30,7 +30,7 @@
  * modulemd_translation_entry_parse_yaml:
  * @parser: (inout): A libyaml parser object positioned at the beginning of a
  * Translation Entry's mapping entry in the YAML document.
- * @locale: (int): A string with the locale for the current translation entry.
+ * @locale: (in): A string with the locale for the current translation entry.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-translation-entry-private.h]: Good catch.

File: modulemd/include/private/modulemd-yaml.h [hunk 4]
```diff
@@ -705,9 +995,43 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
...
+ * @emitter: (inout): A libyaml emitter object positioned where a scalar
+ * belongs in the YAML document.
+ * @error: (out): A #GError that will return the reason for an output error.
+ * @key: (in): The key (string) to be written.
+ * @value: (in): The scalar (string) to be written.
+ * @style: (in): The YAML scalar style for the output.
+ *
+ * Emits key/value pair (@key: @value) using style @style.
+ *
+ * DIRECT USE OF THIS MACRO SHOULD BE AVOIDED. This is part of the internal
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: Same statement as for `EMIT_SCALAR_FULL`

File: modulemd/include/private/modulemd-yaml.h [hunk 5]
```diff
@@ -801,6 +1243,22 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
     }                                                                         \
   while (0)
 
+/**
+ * EMIT_HASHTABLE_KEY_VALUES_IF_NON_EMPTY:
+ * @emitter: (inout): A libyaml emitter object that is positioned at the start
+ * of where a new mapping will be written.
+ * @error: (out): A #GError that will return the reason for any error.
+ * @key: (in): The name to be used as the identifier for the output mapping.
+ * @table: (in): The #GHashTable that is to be output.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: Indicate that this function only works on a `GHashTable` where both the keys and values are strings.

File: modulemd/include/private/modulemd-yaml.h [hunk 6]
```diff
@@ -824,6 +1282,23 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
     }                                                                         \
   while (0)
 
+/**
+ * EMIT_STRING_SET:
+ * @emitter: (inout): A libyaml emitter object that is positioned at the start
+ * of where a new sequence will be written.
+ * @error: (out): A #GError that will return the reason for any error.
+ * @key: (in): The name to be used as the identifier for the output sequence.
+ * @table: (in): The #GHashTable that is to be output.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: Indicate that this `GHashTable` must be a `set()` of string keys.

Also, it outputs the keys, not the values of the table.

File: modulemd/include/private/modulemd-yaml.h [hunk 7]
```diff
@@ -841,6 +1316,23 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
     }                                                                         \
   while (0)
 
+/**
+ * EMIT_STRING_SET_IF_NON_EMPTY:
+ * @emitter: (inout): A libyaml emitter object that is positioned at the start
+ * of where a new sequence will be written.
+ * @error: (out): A #GError that will return the reason for any error.
+ * @key: (in): The name to be used as the identifier for the output sequence.
+ * @table: (in): The #GHashTable that is to be output.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: See `EMIT_STRING_SET`

File: modulemd/include/private/modulemd-yaml.h [hunk 8]
```diff
@@ -851,6 +1343,22 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
     }                                                                         \
   while (0)
 
+/**
+ * EMIT_STRING_SET_FULL:
+ * @emitter: (inout): A libyaml emitter object that is positioned at the start
+ * of where a new sequence will be written.
+ * @error: (out): A #GError that will return the reason for any error.
+ * @key: (in): The name to be used as the identifier for the output sequence.
+ * @table: (in): The #GHashTable that is to be output.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: See `EMIT_STRING_SET`

File: modulemd/include/private/modulemd-yaml.h [hunk 9]
```diff
@@ -867,6 +1375,23 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
     }                                                                         \
   while (0)
 
+/**
+ * EMIT_ARRAY_VALUES:
+ * @emitter: (inout): A libyaml emitter object that is positioned at the start
+ * of where a new sequence will be written.
+ * @error: (out): A #GError that will return the reason for any error.
+ * @key: (in): The name to be used as the identifier for the output sequence.
```
Comment (sgallagher) [inline modulemd/include/private/modulemd-yaml.h]: Indicate that this function outputs both a key and an array value for that key. So it must be run from within a YAML mapping.

Comment (sgallagher) [reply modulemd/include/private/modulemd-yaml.h]: There are several other places that need to note this as well.

**Comment by mmathesius** [comment]
Thanks for the feedback! My latest commit has your comments addressed and has been rebased to current master.

**Pull request merged**